bubblegum/program/tests/utils/tree.rs

```diff
@@ -0,0 +1,631 @@
+        self.verify_creator_tx(args, creator).await?.execute().await
+    }
+
+    pub async fn unverify_creator_tx<'a>(
+        &'a self,
+        args: &'a mut LeafArgs,
+        creator: &Keypair,
+    ) -> Result<UnverifyCreatorBuilder<MAX_DEPTH, MAX_BUFFER_SIZE>> {
+        let root = self.decode_root().await?;
+        let (data_hash, creator_hash) = compute_metadata_hashes(&args.metadata)?;
+
+        let accounts = mpl_bubblegum::accounts::CreatorVerification {
+            tree_authority: self.authority(),
+            leaf_owner: args.owner.pubkey(),
+            leaf_delegate: args.delegate.pubkey(),
+            payer: creator.pubkey(),
+            creator: creator.pubkey(),
+            log_wrapper: spl_noop::id(),
+            compression_program: spl_account_compression::id(),
+            merkle_tree: self.tree_pubkey(),
+        };
+
+        let data = mpl_bubblegum::instruction::UnverifyCreator {
+            root,
+            data_hash,
+            creator_hash,
+            nonce: args.nonce,
+            index: args.index,
+            message: args.metadata.clone(),
+        };
+
+        let need_proof = Some(args.index);
+
+        let inner = CreatorVerificationInner {
+            args,
+            creator_key: creator.pubkey(),
+        };
+
+        Ok(self.tx_builder(
+            accounts,
+            data,
+            need_proof,
+            inner,
+            creator.pubkey(),
+            &[creator],
+        ))
+    }
+
+    pub async fn unverify_creator(&self, args: &mut LeafArgs, creator: &Keypair) -> Result<()> {
+        self.unverify_creator_tx(args, creator)
+            .await?
+            .execute()
+            .await
+    }
+
+    pub async fn transfer_tx<'a>(
+        &'a self,
+        args: &'a mut LeafArgs,
+        new_leaf_owner: &Keypair,
+    ) -> Result<TransferBuilder<MAX_DEPTH, MAX_BUFFER_SIZE>> {
+        let root = self.decode_root().await?;
+        let (data_hash, creator_hash) = compute_metadata_hashes(&args.metadata)?;
+
+        let accounts = mpl_bubblegum::accounts::Transfer {
+            tree_authority: self.authority(),
+            leaf_owner: args.owner.pubkey(),
+            leaf_delegate: args.delegate.pubkey(),
+            new_leaf_owner: new_leaf_owner.pubkey(),
+            log_wrapper: spl_noop::id(),
+            compression_program: spl_account_compression::id(),
+            merkle_tree: self.tree_pubkey(),
+        };
+
+        let data = mpl_bubblegum::instruction::Transfer {
+            root,
+            data_hash,
+            creator_hash,
+            nonce: args.nonce,
+            index: args.index,
+        };
+
+        // Cloning to avoid issues with the borrow checker when passing `&mut args`
+        // to the builder below.
+        let owner = clone_keypair(&args.owner);
+        let new_owner = clone_keypair(new_leaf_owner);
+
+        let need_proof = Some(args.index);
```

Comment:
So with most if not all of these transactions looks like you are essentially hardcoding them to to needing a proof, right?  That is what the `Some()` variant indicates.

So it seems like that is not strictly true for all transfer instructions but is probably true for our default test cases now that there are more than one leaf?  Should there be a mechanism to run it with and without proofs?

Reply:
It's not as much hard-coded as pre-configured :D, but it's meant to be an implementation detail behind the scenes; I can def add some helper method and/or parameter somewhere to make it easier to tweak the value between some and none, and that can easily be added on top at any time. Also, it seems like we need proofs for any tree that has more than just 1 leaf.

Reply:
It could send None if there's only one leaf then, which you know from your proof tree?